web/vim_lib.js

```diff
@@ -72,6 +75,38 @@ var LibraryVIM = {
 
     dropbox: null,
 
+    // from https://github.com/s-macke/jor1k/blob/master/js/lib/utf8.js
+    // LICENSE: https://github.com/s-macke/jor1k/blob/master/LICENSE.md
+    UnicodeToUTF8Stream: function(key) {
```

Comment:
What kind of bugs are they? If they are general, you might report them.
On the other hand, we can also use C code in Vim, if that's not hard. Because the code will be generated in ASM.js, which are likely to be faster than manually written JS.